deflate.c

```diff
@@ -299,11 +299,6 @@ int32_t Z_EXPORT PREFIX(deflateInit2_)(PREFIX3(stream) *strm, int32_t level, int
     if (windowBits == 8)
         windowBits = 9;  /* until 256-byte window bug fixed */
 
-#if !defined(NO_QUICK_STRATEGY) && !defined(S390_DFLTCC_DEFLATE)
```

Comment:
Removing `windowBits = 13` is good in my opinion. It was necessary for `deflate_quick` at the time because it was originally using pre-calculated codes tables and to add the full 32k window would have increased the table size by a lot.

This will also be good for switching between levels. Because if I remember correctly, window size is fixed at start so switching between levels starting at level 1 would have fixed the window size at 8k for all other levels afterwards.

Reply:
Yes, it makes things unnecessarily complicated.